### ISSUE TITLE
Feature request: Infer modid in EBS annotation

### ISSUE THREAD

**Issue by PaintNinja** [GroovyMC/APLPKIGB#5]
Currently APLP supports the same EBS syntax as javafml, which is great for porting and ease-of-use:
```groovy
@Mod.EventBusSubscriber(modid = Groovylicious.MOD_ID, bus = Mod.EventBusSubscriber.Bus.MOD, value = Dist.CLIENT)
```

It would be nice if the modid part was optional:
```groovy
@Mod.EventBusSubscriber(bus = Mod.EventBusSubscriber.Bus.MOD, value = Dist.CLIENT)
```

**Comment by TheSilkMiner** [comment]
[As the documentation of `@EventBusSubscriber` mentions](https://github.com/MinecraftForge/MinecraftForge/blob/1.18.x/javafmllanguage/src/main/java/net/minecraftforge/fml/common/Mod.java#L71), the value is mentioned as "only necessary if this annotation is not on the same class as `@Mod`". Since EBS is mostly used on classes that are not annotated as `@Mod`, this makes the optionality of the field effectively void except for an extremely small percentage of the uses.
For this reason I have decided not to carry over this semantic in APLP's annotation.

**Comment by PaintNinja** [comment]
I meant having the modid part being optional in any class of the mod, including classes that are not annotated as `@Mod`

**Comment by TheSilkMiner** [comment]
According to the documentation, in those instances the field isn't really optional. Currently Forge just guesses what it could be, but it looks like it's not intended behavior according to the docs.

**Comment by PaintNinja** [comment]
I think there's a minor misunderstanding here. This is a feature request for the modid to be optional in APLP's EBS in any class of the mod, not a bug report for an inconsistency with javafml.

**Comment by PaintNinja** [comment]
Implemented in v1.1.0
